compiler/rustc_parse/src/lexer/tokentrees.rs

```diff
@@ -18,36 +18,39 @@ pub(super) struct TokenTreesReader<'a> {
 impl<'a> TokenTreesReader<'a> {
     pub(super) fn parse_all_token_trees(
         string_reader: StringReader<'a>,
-    ) -> (PResult<'a, TokenStream>, Vec<UnmatchedDelim>) {
+    ) -> (TokenStream, PResult<'a, ()>, Vec<UnmatchedDelim>) {
         let mut tt_reader = TokenTreesReader {
             string_reader,
             token: Token::dummy(),
             diag_info: TokenTreeDiagInfo::default(),
         };
-        let res = tt_reader.parse_token_trees(/* is_delimited */ false);
-        (res, tt_reader.diag_info.unmatched_delims)
+        let (stream, res) = tt_reader.parse_token_trees(/* is_delimited */ false);
+        (stream, res, tt_reader.diag_info.unmatched_delims)
     }
 
     // Parse a stream of tokens into a list of `TokenTree`s.
-    fn parse_token_trees(&mut self, is_delimited: bool) -> PResult<'a, TokenStream> {
+    fn parse_token_trees(&mut self, is_delimited: bool) -> (TokenStream, PResult<'a, ()>) {
         self.token = self.string_reader.next_token().0;
         let mut buf = Vec::new();
         loop {
             match self.token.kind {
-                token::OpenDelim(delim) => buf.push(self.parse_token_tree_open_delim(delim)?),
+                token::OpenDelim(delim) => {
```

Comment:
```suggestion
                token::OpenDelim(delim) => buf.push(self.parse_token_tree_open_delim(delim).map_err(....)?),
```

Reply:
The parent item does not return `Result`, it returns `(TokenStream, PResult<'a, ()>, Vec<UnmatchedDelim>)`. I can write an `impl Try` but that seems to be much more code than the somewhat annoying match. Can also use let else instead.